source/slang/slang-check-decl.cpp

```diff
@@ -9745,13 +9748,14 @@ namespace Slang
             ensureDecl(visitor, referencedDecl, DeclCheckState::CapabilityChecked);
         }
         
-        if (resultCaps.implies(nodeCaps))
```

Comment:
So in your example, after processing Store, the resultCaps is (Fragment|Compute). 

When we process Sample, resultCaps shouldn't be implying Compute? So this implies call should return false.

In that case the two capabilities will be joined to form the new resultCaps that is just (Fragment).

So why do we need to remove the implies call here?

My guess is that it is returning true in this case and that is questionable.

Reply:
I actually we can add back the implies now (I think partially what forcing join solved was something another fix I added closer to the end). CI of course will check to be sure.